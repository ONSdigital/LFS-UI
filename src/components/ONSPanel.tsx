import React, { Component } from 'react';

interface Props{
  label: string,
  children: any,
  header?: string,
  status? : string,
  spacious?: boolean,
  id?: string
}

export class ONSPanel extends Component <Props, {}>{

    render() {
      let className = "panel panel--" + (this.props.status === "success" ? "success" : this.props.status === "error" ? "error": "info") + " panel--simple " + (this.props.spacious ? "panel--spacious" : "");
      return (
        <div id={this.props.id} className={className}>
          {this.props.header !== undefined &&
            <div className="panel__header">
              <h1 className="panel__title u-fs-r--b">{this.props.header}</h1>
            </div>
          }
          <div className="panel__body">
              {this.props.children}
            </div>
        </div> 
      );
  }
}
