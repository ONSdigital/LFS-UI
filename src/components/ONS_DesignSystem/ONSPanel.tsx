import React, {Component} from 'react';

interface Props{
  label: string,
  children: any,
  status? : string,
  spacious?: boolean,
  id?: string,
  hidden?: boolean
}

export class ONSPanel extends Component <Props, {}>{

    render() {
      let className = "panel panel--" + (this.props.status === "success" ? "success" : this.props.status === "error" ? "error": "info") + " panel--simple " + (this.props.spacious ? "panel--spacious" : "");
      return (
        <div id={this.props.id} className={className} hidden={this.props.hidden}>
          <div className="panel__body">
              {this.props.children}
          </div>
        </div> 
      );
  }
}
