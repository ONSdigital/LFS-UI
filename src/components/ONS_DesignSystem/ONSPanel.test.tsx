import React from 'react';
import { shallow, mount } from 'enzyme';
import {ONSPanel} from "./ONSPanel";
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

describe("ONS Panel Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const panelProps = {
        label: "Success",
        children: <p>Succceeesssss</p>,
    };

    const statusPanelProps = {
        label: "Status",
        children: <p>Statusssss</p>,
        status: "success",
        // spacious: false
    };

    const spaciousPanelProps = {
        label: "Spacious",
        children: <p>Spaciooouuusssss</p>,
        status: "error",
        spacious: true,
        id: "spacious"
    }
    
    function wrapper (render: any, panelProps: any) {
        return render(
            <ONSPanel   label={panelProps.label} 
                        status={panelProps.status}  
                        spacious={panelProps.spacious} 
                        id={panelProps.id}
                        hidden = {panelProps.hidden}
                        >{panelProps.children}</ONSPanel>)}

    it("should render correctly", () => expect(wrapper(shallow, panelProps).exists()).toEqual(true));

    it("should render the correct label", () => {
        expect(wrapper(mount, panelProps).find("ONSPanel").getElement().props.label).toEqual(panelProps.label);
    });

    it("should render the correct children", () => {
        expect(wrapper(mount, panelProps).find("ONSPanel").getElement().props.children).toEqual(panelProps.children)
    })

    it("should render the correct status", () => {
        expect(wrapper(mount, statusPanelProps).find("div.panel").hasClass('panel--success')).toEqual(true)
    })
    
    it('displays a spacious panel button', () => {
        expect(wrapper(mount, spaciousPanelProps).find('div.panel').hasClass('panel--spacious')).toEqual(true)
    })

})