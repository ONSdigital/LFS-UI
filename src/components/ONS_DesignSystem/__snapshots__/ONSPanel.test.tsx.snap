// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Panel Test matches Snapshot 1`] = `ShallowWrapper {}`;
